st2client/st2client/commands/keyvalue.py

```diff
@@ -77,23 +80,40 @@ def __init__(self, *args, **kwargs):
                                  help='Scope item is under. Example: "user".')
         self.parser.add_argument('-u', '--user', dest='user', default=None,
                                  help='User for user scoped items (admin only).')
+        self.parser.add_argument('-n', '--last', type=int, dest='last',
+                                 default=self.default_limit,
+                                 help=('List N most recent %s. Default is 50.' %
+                                       self.resource_name))
 
-    def run_and_print(self, args, **kwargs):
+    @resource.add_auth_token_to_kwargs_from_cli
+    def run(self, args, **kwargs):
+        # Filtering options
         if args.prefix:
             kwargs['prefix'] = args.prefix
-
         decrypt = getattr(args, 'decrypt', False)
-        kwargs['params'] = {'decrypt': str(decrypt).lower()}
```

Comment:
Ah, that's a bit nasty, should perhaps update that loop which iterates over `kwargs` to skip param.

I would prefer that personally over "kwargs" abuse (with kwargs abuse you don't know what is a query param and what is a something else).